modules/tensor_mechanics/src/kernels/ADDynamicStressDivergenceTensors.C

```diff
@@ -0,0 +1,91 @@
+
+defineADValidParams(
+    ADDynamicStressDivergenceTensors,
+    ADStressDivergenceTensors,
+    params.addClassDescription(
+        "Residual due to stress related Rayleigh damping and HHT time integration terms ");
+    params.addParam<MaterialPropertyName>("zeta",
+                                          0.0,
+                                          "Name of material property or a constant real "
+                                          "number defining the zeta parameter for the "
+                                          "Rayleigh damping.");
+    params.addParam<Real>("alpha", 0, "alpha parameter for HHT time integration");
+    params.addParam<bool>("static_initialization",
+                          false,
+                          "Set to true to get the system to "
+                          "equillibrium under gravity by running a "
+                          "quasi-static analysis (by solving Ku = F) "
+                          "in the first time step"););
+
+template <ComputeStage compute_stage>
+ADDynamicStressDivergenceTensors<compute_stage>::ADDynamicStressDivergenceTensors(
+    const InputParameters & parameters)
+  : ADStressDivergenceTensors<compute_stage>(parameters),
+    _stress_older(adGetMaterialPropertyOlder<RankTwoTensor>(_base_name + "stress")),
+    _stress_old(adGetMaterialPropertyOld<RankTwoTensor>(_base_name + "stress")),
+    _zeta(adGetMaterialProperty<Real>("zeta")),
+    _alpha(adGetParam<Real>("alpha")),
+    _static_initialization(adGetParam<bool>("static_initialization"))
+{
+}
+
+template <ComputeStage compute_stage>
+ADResidual
+ADDynamicStressDivergenceTensors<compute_stage>::computeQpResidual()
+{
+  /**
+   *This kernel needs to be used only if either Rayleigh damping or numerical damping through HHT
+   *time integration scheme needs to be added to the problem thorugh the stiffness dependent damping
+   * parameter _zeta or HHT parameter _alpha, respectively.
+   *
+   * The residual of _zeta*K*[(1+_alpha)vel-_alpha vel_old]+ alpha K [ u - uold] + K u is required
+   * = _zeta*[(1+_alpha)d/dt (Div sigma)-alpha d/dt(Div sigma_old)] +alpha [Div sigma - Div
+   *sigma_old]+ Div sigma
+   * = _zeta*[(1+alpha)(Div sigma - Div sigma_old)/dt - alpha (Div sigma_old - Div sigma_older)/dt]
+   *   + alpha [Div sigma - Div sigma_old] +Div sigma
+   * = [(1+_alpha)*_zeta/dt +_alpha+1]* Div sigma - [(1+2_alpha)*_zeta/dt + _alpha] Div sigma_old +
+   *_alpha*_zeta/dt Div sigma_older
+   */
+
+  ADResidual residual = 0.0;
+  if (_static_initialization && _t == _dt)
+  {
+    // If static inialization is true, then in the first step residual is only Ku which is
+    // stress.grad(test).
+    residual += _stress[_qp].row(_component) * _grad_test[_i][_qp];
+
+    if (_volumetric_locking_correction)
+      residual +=
+          _stress[_qp].trace() / 3.0 * (_avg_grad_test[_i] - _grad_test[_i][_qp](_component));
+  }
+  else if (_dt > 0)
+  {
+    residual +=
```

Comment:
```suggestion
    residual =
```

Reply:
This looks petty, but an assignment might actually be faster than an unnecessary addition to a zero dual number.